morango/api/viewsets.py

```diff
@@ -288,10 +289,12 @@ def perform_destroy(self, transfersession):
         if transfersession.push:
             # dequeue into store and then delete records
             _dequeue_into_store(transfersession)
-            # update database max counters
-            DatabaseMaxCounter.update_fsics(json.loads(transfersession.client_fsic),
-                                            json.loads(transfersession.server_fsic),
-                                            certificates.Filter(transfersession.filter))
+            # update database max counters but use latest fsics on server
+            with transaction.atomic():
+                server_fsic = DatabaseMaxCounter.calculate_filter_max_counters(Filter(transfersession.filter))
+                DatabaseMaxCounter.update_fsics(json.loads(transfersession.client_fsic),
+                                                server_fsic,
+                                                certificates.Filter(transfersession.filter))
```

Comment:
Woohoo! Seems like it would be more DRY, safe, and encapsulated to move this logic directly inside `update_fsics`, though, unless I'm missing something? It should have access to everything it needs (basically just the filter), right?

Reply:
yeah I did this because the way the tests are written, they break when i try putting that inside the function.

Reply:
Ok, if it's hard to change those assumptions in the tests, could you perhaps just open a tech debt issue for now, and we can address it later?